.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateTimePeriodExtractor.cs

```diff
@@ -133,8 +148,9 @@ private IEnumerable<Token> MergeDateWithTimePeriodSuffix(string text, List<Extra
         // Cases like "today after 2:00pm", "1/1/2015 before 2:00 in the afternoon"
         // Valid connector in English for Before include: "before", "no later than", "in advance of", "prior to", "earlier than", "sooner than", "by", "till", "until"...
         // Valid connector in English for After include: "after", "later than"
-        private bool IsValidConnectorForDateAndTimePeriod(string text)
+        private bool IsValidConnectorForDateAndTimePeriod(string text, out int length, bool inPrefix)

```

Comment:
If you need the length as output, I'd suggest returning the length, and return an error code (-1) when false.
You can declare a Constants entry for this. Maybe INVALID_CONNECTOR_CODE.

Reply:
It's OK to rename such method to GetValidConnectorIndexForDateAndTimePeriod
and declare a new IsValidConnectorForDateAndTimePeriod that just checks the returned index or code and still returns a bool.

Reply:
(the length is needed only when the before/after flag is true).
Modified the method to return the length. 
When the method is called, it is checked if the returned length is >=0 (as for the method in the comment below)